akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -26,7 +28,15 @@ private[http2] final case class Http2SubStream(initialHeaders: ParsedHeadersFram
 object Http2Blueprint {
   // format: OFF
   def serverStack(settings: ServerSettings, log: LoggingAdapter): BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] = {
-    httpLayer(settings, log) atop
+    val parserSettings = settings.parserSettings
+    // This is master header parser, every other usage should do .createShallowCopy()
+    // HttpHeaderParser is not thread safe and should not be called concurrently,
+    // the internal trie, however, has built-in protection and will do copy-on-write
+    val masterHttpHeaderParser = HttpHeaderParser.prime(HttpHeaderParser.unprimed(parserSettings, log, { info =>
```

Comment:
Could you use `HttpHeaderParser.apply`, instead? I think it already does the `prime(unprime)` dance.